Calibration/HcalAlCaRecoProducers/src/AlCaIsoTracksProducer.cc

```diff
@@ -208,354 +208,274 @@ AlCaIsoTracksProducer::AlCaIsoTracksProducer(const edm::ParameterSet& iConfig)
+			   << iEvent.id().event() << " Luminosity " 
+			   << iEvent.luminosityBlock() << " Bunch " 
+			   << iEvent.bunchCrossing();
   
-  ///////////////////////////////
-  
-  ///vector of used hits:
-  std::vector<HcalDetId> usedHitsHC;
-  std::vector<int> usedHitsEC;
-  ///
-
-  // main loop over input tracks
-  for (reco::TrackCollection::const_iterator track=tC.begin(); track!=tC.end(); track++) {
-    bool noChargedTracks = true;
-    int itrk1=0;
-    itrk++;
-    double px = track->px();
-    double py = track->py();
-    double pz = track->pz();
-    double ptrack = sqrt(px*px+py*py+pz*pz);
+  //Step1: Get all the relevant containers
+  trigger::TriggerEvent triggerEvent;
+  edm::Handle<trigger::TriggerEvent> triggerEventHandle;
+  iEvent.getByToken(tok_trigEvt_, triggerEventHandle);
+  if (!triggerEventHandle.isValid()) {
+    edm::LogWarning("HcalIsoTrack") << "Cannot access the collection " << labelTriggerEvent_;
+    return;
+  }
+  edm::Handle<edm::TriggerResults> triggerResults;
+  iEvent.getByToken(tok_trigRes_, triggerResults);
+  if (!triggerResults.isValid()) {
+    edm::LogWarning("HcalIsoTrack") << "Cannot access the collection " << labelTriggerResults_;
+    return;
+  }
+
+  edm::Handle<reco::TrackCollection> trkCollection;
+  iEvent.getByToken(tok_genTrack_, trkCollection);
+  if (!trkCollection.isValid()) {
+    edm::LogWarning("HcalIsoTrack") << "Cannot access the collection " << labelGenTrack_;
+    return;
+  }
+  reco::TrackCollection::const_iterator trkItr;
+  edm::Handle<reco::VertexCollection> recVtxs;
+  iEvent.getByToken(tok_recVtx_, recVtxs);  
+  if (!trkCollection.isValid()) {
+    edm::LogWarning("HcalIsoTrack") << "Cannot access the collection " << labelGenTrack_;
+    return;
+  }
+
+  edm::Handle<reco::BeamSpot> beamSpotH;
+  iEvent.getByToken(tok_bs_, beamSpotH);
+  math::XYZPoint leadPV(0,0,0);
+  if (recVtxs->size()>0 && !((*recVtxs)[0].isFake())) {
+    leadPV = math::XYZPoint((*recVtxs)[0].x(),(*recVtxs)[0].y(),
+			    (*recVtxs)[0].z());
+  } else if (beamSpotH.isValid()) {
+    leadPV = beamSpotH->position();
+  }
+  LogDebug("HcalIsoTrack") << "Primary Vertex " << leadPV;
+
+  edm::Handle<EcalRecHitCollection> barrelRecHitsHandle;
+  iEvent.getByToken(tok_EB_, barrelRecHitsHandle);
+  if (!barrelRecHitsHandle.isValid()) {
+    edm::LogWarning("HcalIsoTrack") << "Cannot access the collection " << labelEB_;
+    return;
+  }
+  edm::Handle<EcalRecHitCollection> endcapRecHitsHandle;
+  iEvent.getByToken(tok_EE_, endcapRecHitsHandle);
+  if (!endcapRecHitsHandle.isValid()) {
+    edm::LogWarning("HcalIsoTrack") << "Cannot access the collection " << labelEE_;
+    return;
+  }
+  edm::Handle<HBHERecHitCollection> hbhe;
+  iEvent.getByToken(tok_hbhe_, hbhe);
+  if (!hbhe.isValid()) {
+    edm::LogWarning("HcalIsoTrack") << "Cannot access the collection " << labelHBHE_;
+    return;
+  }
+
+  //Get L1 trigger object
+  ptL1 = etaL1 = phiL1 = 0;
```

Comment:
`ptL1, etaL1, phiL1` are data members used for producing collections and selecting on them, which are reset at every event in the `produce` function member. Anyhow, they are not initialised in the c'tor. Can you do that?